Spring-Boot/Spring-Boot-Exam/src/test/java/tw/waterball/judgegirl/springboot/exam/controllers/HomeworkControllerTest.java

```diff
@@ -0,0 +1,133 @@
+package tw.waterball.judgegirl.springboot.exam.controllers;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.ResultActions;
+import tw.waterball.judgegirl.entities.problem.Problem;
+import tw.waterball.judgegirl.entities.stubs.ProblemStubs;
+import tw.waterball.judgegirl.examservice.domain.usecases.homework.CreateHomeworkUseCase;
+import tw.waterball.judgegirl.problemapi.clients.FakeProblemServiceDriver;
+import tw.waterball.judgegirl.problemapi.views.ProblemView;
+import tw.waterball.judgegirl.springboot.exam.SpringBootExamApplication;
+import tw.waterball.judgegirl.springboot.exam.view.HomeworkView;
+import tw.waterball.judgegirl.springboot.profiles.Profiles;
+import tw.waterball.judgegirl.submissionapi.clients.SubmissionServiceDriver;
+import tw.waterball.judgegirl.testkit.AbstractSpringBootTest;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ActiveProfiles(Profiles.JWT)
+@ContextConfiguration(classes = SpringBootExamApplication.class)
+public class HomeworkControllerTest extends AbstractSpringBootTest {
+
+    private static final String HOMEWORK_NAME = "homeworkName";
+    private static final String HOMEWORK_PATH = "/api/homework";
+
+    @Autowired
+    private FakeProblemServiceDriver problemServiceDriver;
+
+    @MockBean
+    private SubmissionServiceDriver submissionServiceDriver;
+
+    @AfterEach
+    void cleanUp() {
+        problemServiceDriver.clear();
+    }
+
+    @Test
+    public void GivenThreeProblemsCreated_WhenCreateHomeworkWithProblemIds_ShouldCreateSuccessfully() throws Exception {
+        Integer[] createProblemIds = {1, 2, 3};
+        createProblems(createProblemIds);
+
+        HomeworkView homework = getBody(createHomework(HOMEWORK_NAME, createProblemIds)
+                .andExpect(status().isOk()), HomeworkView.class);
+
+        List<String> homeworkProblemIds = homework.problemIds;
+        assertEquals(createProblemIds.length, homeworkProblemIds.size());
+        for (int i = 0; i < homeworkProblemIds.size(); i++) {
+            assertEquals(String.valueOf(createProblemIds[i]), homeworkProblemIds.get(i));
+        }
```

Comment:
extract a method `homeworkShouldIncludeProblemIds(homework, problemIds)` to make this assertion one-line